Formula/v/vineyard.rb

```diff
@@ -28,55 +29,48 @@ class Vineyard < Formula
-    sha256 "01a1e793faa5bd89abc851fa15d0a0db26f160890c7102cd8dce643e886b47f5"
+    url "https://files.pythonhosted.org/packages/27/b8/f21073fde99492b33ca357876430822e4800cdf522011f18041351dfa74b/setuptools-75.1.0.tar.gz"
+    sha256 "d59a21b17a275fb872a9c3dae73963160ae079f1049ed956880cd7c09b120538"
   end
 
   def install
-    python = "python3.12"
-    venv = virtualenv_create(libexec, python)
+    python3 = "python3.12"
+    venv = virtualenv_create(buildpath/"venv", python3)
     venv.pip_install resources
     # LLVM is keg-only.
-    ENV.prepend_path "PYTHONPATH", llvm.opt_prefix/Language::Python.site_packages(python)
+    llvm = deps.map(&:to_formula).find { |f| f.name.match?(/^llvm(@\d+)?$/) }
+    ENV.prepend_path "PYTHONPATH", llvm.opt_prefix/Language::Python.site_packages(python3)
 
     # Work around an Xcode 15 linker issue which causes linkage against LLVM's
     # libunwind due to it being present in a library search path.
-    ENV.remove "HOMEBREW_LIBRARY_PATHS", llvm.opt_lib if DevelopmentTools.clang_build_version >= 1500
-    ENV.append "LDFLAGS", "-Wl,-rpath,#{llvm.opt_lib}/#{Hardware::CPU.arch}-unknown-linux-gnu" if OS.linux?
-
-    system "cmake", "-S", ".", "-B", "build",
-                    "-DCMAKE_CXX_STANDARD=17",
-                    "-DCMAKE_CXX_STANDARD_REQUIRED=TRUE",
-                    "-DPYTHON_EXECUTABLE=#{venv.root}/bin/python",
-                    "-DUSE_EXTERNAL_ETCD_LIBS=ON",
-                    "-DUSE_EXTERNAL_REDIS_LIBS=ON",
-                    "-DUSE_EXTERNAL_HIREDIS_LIBS=ON",
-                    "-DBUILD_VINEYARD_TESTS=OFF",
-                    "-DUSE_LIBUNWIND=OFF",
-                    "-DLIBGRAPELITE_INCLUDE_DIRS=#{Formula["libgrape-lite"].opt_include}",
-                    "-DOPENSSL_ROOT_DIR=#{Formula["openssl@3"].opt_prefix}",
-                    "-DBUILD_VINEYARD_SERVER_ETCD=OFF", # Fails with protobuf 27
-                    *std_cmake_args
+    if DevelopmentTools.clang_build_version >= 1500
+      recursive_dependencies
+        .select { |d| d.name.match?(/^llvm(@\d+)?$/) }
+        .map { |llvm_dep| llvm_dep.to_formula.opt_lib }
+        .each { |llvm_lib| ENV.remove "HOMEBREW_LIBRARY_PATHS", llvm_lib }
+    end
+
+    args = [
+      "-DBUILD_VINEYARD_PYTHON_BINDINGS=OFF",
+      "-DBUILD_VINEYARD_TESTS=OFF",
+      "-DCMAKE_CXX_STANDARD=17",
+      "-DCMAKE_CXX_STANDARD_REQUIRED=TRUE",
+      "-DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON", # for newer protobuf
+      "-DLIBGRAPELITE_INCLUDE_DIRS=#{Formula["libgrape-lite"].opt_include}",
+      "-DOPENSSL_ROOT_DIR=#{Formula["openssl@3"].opt_prefix}",
+      "-DPYTHON_EXECUTABLE=#{venv.root}/bin/python",
+      "-DUSE_EXTERNAL_ETCD_LIBS=ON",
+      "-DUSE_EXTERNAL_HIREDIS_LIBS=ON",
+      "-DUSE_EXTERNAL_REDIS_LIBS=ON",
+      "-DUSE_LIBUNWIND=OFF",
+    ]
+    args << "-DCMAKE_EXE_LINKER_FLAGS=-Wl,-dead_strip_dylibs"
```

Comment:
Will need to add OS check:
```suggestion
    args << "-DCMAKE_EXE_LINKER_FLAGS=-Wl,-dead_strip_dylibs" if OS.mac?
```